tests/extension/oneapi_device_global/device_global_handler_copy_memcpy_overloads.cpp

```diff
@@ -0,0 +1,424 @@
+        // dev_global have to be equal to *dest after copy
+        is_copy_correct_acc[0] =
+            value_helper<T>::compare_val(dev_global<T>, *dest);
+      });
+    });
+    queue.wait_and_throw();
+
+    // If T is array we can test .copy() method with count and startIndex
+    // parameters
+    if constexpr (elements_count > 1) {
+      queue.submit([&](sycl::handler& cgh) {
+        auto is_copy_correct_acc =
+            is_copy_correct_buf.template get_access<sycl::access_mode::write>(
+                cgh);
+
+        sycl_stub::handler cgh_stub;
+        // Changing value of the first element
+        dev_global<T>[0] = changed_value;
+        // Copy first element from the array
+        cgh_stub.template copy<T>(dev_global<T>, dest, element_size, 0);
+        cgh.single_task<kernel2<T>>([=] {
+          // Compare again after copy
+          is_copy_correct_acc[0] &=
+              value_helper<T>::compare_val(dev_global<T>, *dest);
+        });
+      });
+      queue.wait_and_throw();
+    }
+  }
+
+  if (!is_copy_correct) {
+    FAIL(log, get_case_description(
+                  "deivice_global: sycl::handler .copy() method overload",
+                  "Wrong value after copy from the device_global instance",
+                  type_name));
+  }
+}
+}  // namespace copy_from_dg
+
+namespace memcpy_to_dg {
+template <typename T>
+struct kernel1;
+template <typename T>
+struct kernel2;
+// Creating instance with default constructor
+template <typename T>
+oneapi::device_global<T> dev_global;
+
+/** @brief The function tests that .memcpy() method overload correctly copy
+ * memory from the pointer to the device_global instance
+ *  @tparam T Type of underlying device_global value
+ */
+template <typename T>
+void run_test(util::logger& log, const std::string& type_name) {
+  using element_type = std::remove_all_extents_t<T>;
+  using element_ptr = element_type*;
+  using void_ptr = void*;
+
+  // Count size of one element
+  constexpr size_t element_size = sizeof(element_type);
+  // Count number of elements
+  constexpr size_t elements_count = sizeof(T) / element_size;
+
+  element_type init_value;
+  init_value = 1;
+  element_type changed_value;
+  changed_value = 2;
+
+  T src_value;
+  value_helper<T>::change_val(src_value, init_value);
+  void_ptr src = static_cast<void_ptr>(pointer_helper(src_value));
+
+  bool is_copy_correct{false};
+  auto queue = util::get_cts_object::queue();
+  {
+    sycl::buffer<bool, 1> is_copy_correct_buf(&is_copy_correct,
+                                              sycl::range<1>(1));
+    auto queue = util::get_cts_object::queue();
+    queue.submit([&](sycl::handler& cgh) {
+      auto is_copy_correct_acc =
+          is_copy_correct_buf.template get_access<sycl::access_mode::write>(
+              cgh);
+
+      sycl_stub::handler cgh_stub;
+      // Copy memory from the pointed memory to the device_global instance
+      cgh_stub.template memcpy<T>(dev_global<T>, src);
```

Comment:
Do we need all these `template`s?

Reply:
Well, I found restriction in [C++ standard](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2012/n3337.pdf) in paragraph 14.2/4. It is an old standard but an example with ill-formed expression case an error even on C++17, but I can't find the same article in the C++17 standard.

According to this document, we have to use the `template` prefix when "T is not a member of
the current instantiation", which is not our case, because `handler` is not templated. So I will remove the `template` keyword from `memcpy` and `copy` invocations